repo/repo.go

```diff
@@ -122,9 +136,23 @@ func ListRepositories(workspace *rule.File) (repos []*rule.Rule, repoFileMap map
 				}
 				repoFileMap[repo.Name()] = macroFile
 			}
+
+			// We also want to double check if there are more repository_macro
+			// directives within this repository_macro
+			repos, repoIndexMap, repoFileMap, err = getRepositoriesInside(workspace, f, repos, repoIndexMap, repoFileMap)
 		}
 	}
-	return repos, repoFileMap, nil
+	return repos, repoIndexMap, repoFileMap, nil
+}
+
+// Since directives should not be within the macro file's defined function, the file
```

Comment:
Why shouldn't directives be within a macro function? To me it should probably be the alternative, directives _must_ be within the macro function. Otherwise, a directive would be double-counted for the case of multiple macros within the same file.